### PR TITLE
Task 1/vulkan basic initialization

### PR DESCRIPTION
**Pull request by krupitskas**

### CODE REVIEW

File: VorpalEngine/include/vorpal/VorpalEngine.hpp [hunk 1]
```diff
@@ -1,9 +1,10 @@
+// 73s studio
+
 #ifndef VORPAL_VORPAL_ENGINE_HPP
 #define VORPAL_VORPAL_ENGINE_HPP
 
-#include <vorpal/utility.hpp>
-
-struct GLFWwindow;
+#include <vorpal/core/Utility.hpp>
+#include <vorpal/video/Graphics.hpp>
```
Comment (dsodintsov) [inline VorpalEngine/include/vorpal/VorpalEngine.hpp]: use forward declaration

File: VorpalEngine/include/vorpal/VorpalEngine.hpp [hunk 2]
```diff
@@ -1,9 +1,10 @@
+// 73s studio
+
 #ifndef VORPAL_VORPAL_ENGINE_HPP
 #define VORPAL_VORPAL_ENGINE_HPP
 
-#include <vorpal/utility.hpp>
-
-struct GLFWwindow;
+#include <vorpal/core/Utility.hpp>
```
Comment (dsodintsov) [inline VorpalEngine/include/vorpal/VorpalEngine.hpp]: why is this requested here? it is not used in this .hpp

File: VorpalEngine/include/vorpal/video/Graphics.hpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// 73s studio
+
+#ifndef VORPAL_RENDERER_HPP
+#define VORPAL_RENDERER_HPP
+
+class Renderer;
```
Comment (dsodintsov) [inline VorpalEngine/include/vorpal/video/Graphics.hpp]: should this be in vp namespace?

File: VorpalEngine/include/vorpal/VorpalEngine.hpp [hunk 4]
```diff
@@ -1,29 +1,32 @@
-#ifndef VORPAL_VORPAL_ENGINE_HPP
-#define VORPAL_VORPAL_ENGINE_HPP
+#ifndef VORPAL_ENGINE_HPP
+#define VORPAL_ENGINE_HPP
 
-#include <vorpal/utility.hpp>
+#include <string>
```
Comment (MrDmitry) [inline VorpalEngine/include/vorpal/VorpalEngine.hpp]: why?

File: VorpalEngine/include/vorpal/core/Logger.hpp [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#ifndef VORPAL_LOGGER_HPP
+#define VORPAL_LOGGER_HPP
+
+#include <loguru.hpp>
+
+#define LOG_ERROR(...)    LOG_F(ERROR, __VA_ARGS__)
+#define LOG_WARNING(...)  LOG_F(WARNING, __VA_ARGS__)
+#define LOG_INFO(...)     LOG_F(INFO, __VA_ARGS__)
+#define LOG_DEBUG(...)    DLOG_F(INFO, __VA_ARGS__)
+#endif //VORPAL_LOGGER_HPP
```
Comment (MrDmitry) [inline VorpalEngine/include/vorpal/core/Logger.hpp]: space after // ?

File: VorpalEngine/include/vorpal/core/Settings.hpp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  Settings();
+  static Settings *s_settings;
+
+  unsigned int m_width, m_height;
+  std::string m_applicationName, m_vorpalEngineName;
+};
+}
+}
+
+#endif //VORPAL_SETTINGS_HPP
```
Comment (dsodintsov) [inline VorpalEngine/include/vorpal/core/Settings.hpp]: newline at the end

File: VorpalEngine/include/vorpal/core/Timer.hpp [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace vp {
+namespace core {
+class Timer {
+
+};
+}
+}
+
+#endif //VORPAL_TIMER_HPP
```
Comment (dsodintsov) [inline VorpalEngine/include/vorpal/core/Timer.hpp]: newline at the end

File: VorpalEngine/include/vorpal/core/Settings.hpp (hunk 6 above)
Comment (MrDmitry) [inline VorpalEngine/include/vorpal/core/Settings.hpp:38]: space after //

File: VorpalEngine/include/vorpal/core/Timer.hpp (hunk 7 above)
Comment (MrDmitry) [inline VorpalEngine/include/vorpal/core/Timer.hpp:14]: space after //

**Review by MrDmitry — APPROVED** [review]
w/e style will be done separately

**Pull request merged**